lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -149,20 +178,31 @@ def filter_recommended_courses(user, unfiltered_course_keys, recommendation_coun
             - marketing url for the course (marketing_url)
     """
     filtered_recommended_courses = []
-    fields = ["title", "owners", "marketing_url"]
+    fields = ["uuid", "title", "owners", "image", "url_slug", "course_runs", "location_restriction"]
 
     # Remove the course keys a user is already enrolled in
-    enrollable_course_keys = _remove_user_enrolled_course_keys(user, unfiltered_course_keys)[:recommendation_count]
+    enrollable_course_keys = _remove_user_enrolled_course_keys(user, unfiltered_course_keys)
+    # Remove the course which is user currently visiting
+    recommended_course_keys = [course_key for course_key in enrollable_course_keys if course_key != visiting_course]
 
-    for course_id in enrollable_course_keys:
+    for course_id in recommended_course_keys[:recommendation_count]:
         course_data = get_course_data(course_id, fields)
-        if course_data:
+        if course_data and not _has_product_country_restriction(course_data, country_code):
+            active_course_run = [course_run for course_run in course_data.get('course_runs')
+                                 if course_run.get('availability') == 'Current'][0]
+
             filtered_recommended_courses.append(
                 {
-                    "course_key": course_id,
-                    "title": course_data["title"],
-                    "logo_image_url": course_data["owners"][0]["logo_image_url"],
-                    "marketing_url": course_data.get("marketing_url"),
+                    "uuid": course_data.get("uuid"),
+                    "title": course_data.get("title"),
+                    "image": course_data.get("image"),
+                    "owners": course_data.get("owners"),
+                    "prospectusPath": f"courses/{course_data.get('url_slug')}",
+                    "activeCourseRun": {
+                        "key": active_course_run.get("key"),
+                        "type": "Active",
+                        "marketingUrl": active_course_run.get("marketing_url"),
+                    },
```

Comment:
This same function is being used by learner dashboard on both MFE and legacy view. This change will break it.